aicsimageio/writers/ome_zarr_writer.py

```diff
@@ -249,26 +264,32 @@ def write_image(
             * image_data.shape[ydimindex]
             * image_data.itemsize
         )
+
         target_chunk_size = 16 * (1024 * 1024)  # 16 MB
         # this is making an assumption of chunking whole XY planes.
         # TODO allow callers to configure chunk dims?
         nplanes_per_chunk = int(math.ceil(target_chunk_size / plane_size))
         nplanes_per_chunk = (
             min(nplanes_per_chunk, image_data.shape[zdimindex]) if zdimindex > -1 else 1
         )
-        chunk_dim_map = {
-            "T": 1,
-            "C": 1,
-            "Z": nplanes_per_chunk,
-            "Y": image_data.shape[ydimindex],
-            "X": image_data.shape[xdimindex],
-        }
-        chunk_dims = [
+
+        if chunk_dims is None:
+            chunk_dim_map = {
+                "T": 1,
+                "C": 1,
+                "Z": nplanes_per_chunk,
+                "Y": image_data.shape[ydimindex],
+                "X": image_data.shape[xdimindex],
+            }
+            chunk_dims = tuple(chunk_dim_map[d] for d in dimension_order)
```

Comment:
Your `build_chunk_dims` method is unused, looks like maybe you meant to use it here?